kotlintest-assertions/src/main/kotlin/io/kotlintest/matchers/result/matchers.kt

```diff
@@ -0,0 +1,53 @@
+package io.kotlintest.matchers.result
+
+import io.kotlintest.Matcher
+import io.kotlintest.should
+import io.kotlintest.shouldNot
+
+fun <T> Result<T>.shouldBeSuccess() = this should beSuccess()
```

Comment:
We should add in contracts for these too and lambda versions.
You can look at the arrow Validation and Either matchers to see how it's done.

Reply:
Do you mean [this](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.contracts/index.html)? If so, how can we do this here? With Kotlin Result, we don't have a `Success` and `Error` class like on Arrow classes. And to get the value, Result uses a function and we cannot make a "not null" implication.

Reply:
Do you mean [this](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.contracts/index.html)? If so, how can we do this here? With Kotlin Result, we don't have a `Success` and `Error` class like on Arrow classes. And to get the value, Result uses a function and we cannot make a "not null" implication.

Reply:
Sorry, quite right.
Contracts make no sense.

We can do the lambda version though,

result.shouldbeSuccess { println("My success $it" }

Reply:
Done! Also added the items do to matchers doc.